recorder/src/test/java/org/arquillian/cube/q/recorder/TakeNetworkChaosInformationTest.java

```diff
@@ -0,0 +1,124 @@
+        ToxicCreated toxicCreated = new ToxicCreated(timeout);
+
+        TakeNetworkChaosInformation takeNetworkChaosInformation = new TakeNetworkChaosInformation();
+        ArrayList<Map<String, Object>> toxics = getToxicList(takeNetworkChaosInformation);
+
+        //when
+        takeNetworkChaosInformation.captureToxicDetailsAfterCreate(toxicCreated, toxicAction);
+
+        //then
+        assertThat(toxics).hasSize(1);
+        assertThat(toxics.get(0)).containsExactly(entry("actionon", "helloworld"),
+                entry("type", "Timeout"),
+                entry("phase", "create"),
+                entry("toxic", timeout)
+        );
+    }
+
+    @Test
+    public void should_capture_toxic_details_after_update() throws NoSuchFieldException, IllegalAccessException {
+        //given
+        ToxiProxyClient.Bandwidth bandwidth = new ToxiProxyClient.Bandwidth("bandwidth_downstream", "DOWNSTREAM", 1.0f, rate(1000));
+        ToxicUpdated toxicUpdated = new ToxicUpdated(bandwidth);
+
+        TakeNetworkChaosInformation takeNetworkChaosInformation = new TakeNetworkChaosInformation();
+        ArrayList<Map<String, Object>> toxics = getToxicList(takeNetworkChaosInformation);
+
+        //when
+        takeNetworkChaosInformation.captureToxicDetailsAfterUpdate(toxicUpdated, toxicAction);
+
+        //then
+        assertThat(toxics).hasSize(1);
+        assertThat(toxics.get(0)).containsExactly(
+                entry("actionon", "helloworld"),
+                entry("type", "Bandwidth"),
+                entry("phase", "update"),
+                entry("toxic", bandwidth)
+        );
+    }
+
+    @Test
+    public void should_report_json_file_with_toxicity_params_after_test() throws NoSuchMethodException, IOException {
+
+        TakeNetworkChaosInformation takeNetworkChaosInformation = new TakeNetworkChaosInformation();
+        takeNetworkChaosInformation.propertyReportEvent = propertyReportEvent;
+
+        takeNetworkChaosInformation.reportToxicConfiguration(new After(TakeNetworkChaosInformationTest.class, TakeNetworkChaosInformationTest.class.getMethod("should_report_json_file_with_toxicity_params_after_test")), new ReporterConfiguration());
+
+        verify(propertyReportEvent).fire(propertyReportEventArgumentCaptor.capture());
+
+        final PropertyReportEvent propertyReportEvent = propertyReportEventArgumentCaptor.getValue();
+        final PropertyEntry propertyEntry = propertyReportEvent.getPropertyEntry();
+        assertThat(propertyEntry).isInstanceOf(GroupEntry.class);
+
+        List<PropertyEntry> childEntry = propertyEntry.getPropertyEntries();
+        assertThat(childEntry).hasSize(1).extracting("class.simpleName").containsExactly("FileEntry");
+
+        FileEntry fileEntry = (FileEntry) childEntry.get(0);
+        assertThat(fileEntry.getPath()).isEqualTo("reports/chaos/should_report_json_file_with_toxicity_params_after_test.json");
```

Comment:
Since there are no toxics reported for this isolated test we never really invoke json serialization here. I believe we need more tests for this.